[website] Add why Design Engineer for data grid (#40016)
mui/material-ui: docs/pages/careers/design-engineer-x-grid.md

## Why we're hiring

> [According to our last survey](https://mui.com/blog/2021-developer-survey-results/#what-are-your-most-important-criteria-for-choosing-a-ui-library), the quality of the design execution (UX & UI) is the most important criterion used by developers & designers to pick their next general-purpose UI library.

Reviewer: ```suggestion
[According to our last survey](https://mui.com/blog/2021-developer-survey-results/#what-are-your-most-important-criteria-for-choosing-a-ui-library), the quality of the design execution (UX & UI) is the most important criterion used by developers & designers to choose a component UI library.
```

Slight wording change — "general-purpose" felt too vague.

> We hear an even bigger pain around _design_ from developers who adopt projects specialized into one UI component, e.g. data grid. For them, it's not only the default behaviors that need to be great but also the ease of customizing the components to their needs and integrating into the rest of their application.

Reviewer: ```suggestion
It's an even bigger problem when developers work on projects around a single major component, like the Data Grid. It's not just the interaction that needs to feel great but the developer experience of working with that component also needs to be excellent, to allow a smooth integration with the rest of their app.
```

Also a slight re-phrasing but preserving the general idea.

> Up until now, we solved a subset of these problems by specializing in a MUI X Data Grid that deeply integrates with Material UI.

Reviewer: ```suggestion
Up until now, we're solving a subset of these problems by smoothly integrating the MUI X Data Grid with Material UI.
```

"in a MUI X Data Grid" feels weird to say, but I think it sounds better if we broaden it up a bit! Also, I think putting the verb in the present tense makes slightly more sense, to give leeway that it's a continuous process of tackling this issue rather than something that's closed &done.

> We feel that the timing is right to take a step further and invest in the design of MUI X to have the best data grid not only for Material UI but for **any** React project.

Author: Strategically: It means making it an entry point to the ecosystem. The goal is that once a developer has enough entry points with us, he might decide to adopt everything else, e.g. Microsoft Team vs. Notion

> We feel that the timing is right to take a step further and invest in the design of MUI X to have the best data grid not only for Material UI but for **any** React project.

Reviewer: ```suggestion
But the time has come to take a step further and double-down on design by expanding the Data Grid capacities to integrate nicely with _any_ React project rather than just a Material UI-based one.
```

> The MUI X Data Grid team has no dedicated designer, so we are opening this role to give us the means of this objective.

Reviewer 2: ```suggestion

```

I would remove this. My argument is that this sentence puts the person in a state of mind that they should be a designer, which we discussed that it's not the goal, but even if we keep it I'm unsure how much it will affect people's decision to apply (or not).

Reviewer 3: +1

Reviewer: + 1, agreed! I think the rest of the paragraph already conveys the justification for the role, no need for "so we are opening this role...".
